prow/entrypoint/run.go

```diff
@@ -134,11 +150,47 @@ func (o Options) ExecuteProcess() (int, error) {
 		}
 	}
 
-	executable := o.Args[0]
+	combinedCommandAndArgs := o.CombinedCommandAndArgs()
+	// Set the first arg as entrypoint for now
+	executable := combinedCommandAndArgs[0]
 	var arguments []string
-	if len(o.Args) > 1 {
-		arguments = o.Args[1:]
+	if len(combinedCommandAndArgs) > 1 {
+		arguments = combinedCommandAndArgs[1:]
+	}
+	// Discover default entrypoint from the image only when it's not explicitly
+	// defined in the job
+	if o.UseDefaultEntrypoint != nil && *o.UseDefaultEntrypoint && len(o.Command) == 0 && o.ImageRef != "" {
+		// By default the entrypoint is discovered from "linux/amd64", which
+		// could be overriden by remote.WithPlatform, which probably is going to
+		// be quite some hacks to make it right as mentioned at
+		// https://github.com/kubernetes/test-infra/pull/25383#pullrequestreview-891109864,
+		// so for now we only support either "linux/amd64" or other arches that
+		// share the same entrypoint with its "linux/amd64" variant.
+		img, err := remote.Image(o.ImageRef,
```

Comment:
This just downloads some metadata, not the actual image, correct?

Reply:
Correct, the `img` is just a struct containing metadata like layer blob digests (not contents). When you call `img.ConfigFile` it reuses the creds it used to fetch the image manifest to fetch the config blob (again, just a small JSON blob, no image contents).

Only if you call `img.Layers()` and consume any of the resulting layers' `Compressed()` or `Uncompressed()` `ReadCloser`s will you actually fetch any of the layer data. Otherwise it'll only fetch small JSON blobs for the manifest and image config.